pkg/metrics/listener.go

```diff
@@ -30,9 +30,11 @@ func NewHTTPListener(addr string) error {
 
 	log.L.Infof("Start metrics HTTP server on %s", addr)
 
-	if err := http.ListenAndServe(addr, nil); err != nil {
-		return fmt.Errorf("error serve on %s: %v", addr, err)
-	}
+	go func() {
+		if err := http.ListenAndServe(addr, nil); err != nil {
+			log.L.Errorf("error serve on %s: %v", addr, err)
```

Comment:
Could we output information such as "Failed to serve metrics HTTP endpoint on %s: %v" as this error is in the go routine.

Reply:
Good point. We'd better give more information about error context.